domains/middleware/authorization.go

```diff
@@ -58,6 +58,11 @@ func (am *authorizationMiddleware) CreateDomain(ctx context.Context, session aut
 }
 
 func (am *authorizationMiddleware) RetrieveDomain(ctx context.Context, session authn.Session, id string, withRoles bool) (domains.Domain, error) {
+	if err := am.checkSuperAdmin(ctx, session.UserID); err == nil {
+		session.SuperAdmin = true
+		return am.svc.RetrieveDomain(ctx, session, id, withRoles)
+	}
```

Comment:
Could you please explain why this check is needed here

Reply:
I had included this because we have retriueing the domain with superadmin context here:
https://github.com/absmach/supermq/blob/878860d249d79765399b220f1f8bda8f3d7ab699/domains/service.go#L103-L113

Reply:
Got it